### PR TITLE
feat: update FBCA spec to support VDM v2 events

### PR DESCRIPTION
**Pull request by sandeepdsvs**
## What are the changes introduced in this PR?

Write a brief explainer on your code changes.

## What is the related Linear task?

Resolves INT-XXX

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3675/test-report.html

File: src/v0/destinations/fb_custom_audience/recordTransform.js [hunk 1]
```diff
@@ -100,18 +100,11 @@ const processRecordEventArray = (
...
-    accessToken,
-    disableFormat,
-    type,
-    subType,
-    isRaw,
-    maxUserCount,
-    audienceId,
-  } = destination.Config;
+  const { accessToken, disableFormat, type, subType, isRaw, maxUserCount } = destination.Config;
+  const audienceId = get(connection, 'config.destination.audienceId');
```
Comment (koladilip) [inline src/v0/destinations/fb_custom_audience/recordTransform.js:106]: why do we need to use get here?
what about 
connection?.config?.destination?.audienceId

File: src/v0/destinations/fb_custom_audience/recordTransform.js [hunk 2]
```diff
@@ -100,18 +100,11 @@ const processRecordEventArray = (
...
+  const isHashRequired = get(connection, 'config.destination.isHashRequired');
```
Comment (koladilip) [inline src/v0/destinations/fb_custom_audience/recordTransform.js:107]: same here too

File: src/v0/destinations/fb_custom_audience/util.js [hunk 3]
```diff
@@ -49,14 +49,13 @@ const batchingWithPayloadSize = (payload) => {
 };
 
 const getSchemaForEventMappedToDest = (message) => {
-  const mappedSchema = get(message, 'context.destinationFields');
+  const mappedSchema = get(message, 'fields');
```
Comment (koladilip) [inline src/v0/destinations/fb_custom_audience/util.js:52]: we shouldn't use get when we know the exact path.

message?.fields should be preferred

File: src/v0/destinations/fb_custom_audience/recordTransform.js [hunk 4]
```diff
@@ -136,7 +137,11 @@ async function processRecordInputs(groupedRecordInputs) {
   // user schema validation
   let { userSchema } = destination.Config;
   if (mappedToDestination) {
```
Comment (koladilip) [inline src/v0/destinations/fb_custom_audience/recordTransform.js:139]: are we sending mappedToDestination in new format?

**Comment by koladilip** [comment]
@sandeepdsvs update this PR with new changes

File: test/integrations/destinations/fb_custom_audience/router/batchingRecord.ts [hunk 5]
```diff
@@ -92,6 +106,13 @@ export const rETLBatchingRouterRequest: RouterTransformationRequest = {
       metadata: generateMetadata(2),
     },
     {
+      connection: {
+        config: {
+          destination: {
+            isHashRequired: true,
```
Comment (koladilip) [inline test/integrations/destinations/fb_custom_audience/router/batchingRecord.ts:112]: schema version not required here?

**Comment by koladilip** [comment]
closing for https://github.com/rudderlabs/rudder-transformer/pull/3729

**Pull request closed without merging**